### PR TITLE
Add class naming conventions styleguide

### PR DESCRIPTION
**Pull request by acacheung**

### CODE REVIEW

**Comment by acacheung** [comment]
@NetOperatorWibby can you look over this? Any additions or clarifications needed? Do you think there are any additional common patterns that should be included?

**Comment by NetOperatorWibby** [comment]
@acacheung I think code examples would be good to have, along with the pictures. The images give meaning to the text, but code provides clarity.

File: uxd/class-naming-conventions.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+## Suffixes
+
+* `--r` for right side. Example: `.l-col--r` would style a column that
+  often would be positioned on the right side of a page. This does not
+  have to be true when the screen narrows down to mobile phone width.
+* `--l` for left side.
+* `--s` for size small.
+* `--m` for size medium.
+* `--b` for size big (didn&rsquo;t want it to be `--l`, which would
+  be confused with left).
```
Comment (acacheung) [inline uxd/class-naming-conventions.md]: what are these sizes referring to? examples plz

**Comment by brzpegasus** [comment]
Not directly related to this particular document since you're focusing solely on class naming conventions here, but are you thinking of adding another guide for Sass file structuring (e.g. SMACSS-style base, layout, module, etc. or whatever variation we're adopting at DockYard)?

**Comment by acacheung** [comment]
@brzpegasus yup!

File: uxd/class-naming-conventions.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+[CSS Wizardry&rsquo;s &ldquo;Pain in the class&ldquo; blog post]
+(http://csswizardry.com/2011/09/when-using-ids-can-be-a-pain-in-the-class/).
+Only use IDs for
+[fragment identifiers]
+(http://csswizardry.com/2011/06/namespacing-fragment-identifiers/#fragment-identifiers).
+
+## BEM Naming Conventions
+
+Use `.block__element--modifier` syntax for naming classes. Refer to
+[CSS WIzardry&rsquo;s MindBEMding blog post]
```
Comment (NetOperatorWibby) [inline uxd/class-naming-conventions.md]: "I" is capitalized

File: uxd/class-naming-conventions.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  [SMACSS state rules]
+  (https://smacss.com/book/type-state).
+  Most common use cases: is-active (Example: `.nav__link.is-active`
+  when a nav link is highlighted to show that it is the current page)
+  and is-hidden (hides content visually, but allows screen readers like
+  VoiceOver to still read it. There are many different ways of hiding
+  text, but so far below is the way that works in the most cases even
+  though there is a
+  [performance hit]
+  (http://www.zeldman.com/2012/03/01/replacing-the-9999px-hack-new-image-replacement/)).
```
Comment (NetOperatorWibby) [inline uxd/class-naming-conventions.md]: Why not use what was suggested in the blog post?

```
.hide-text {
  text-indent: 100%;
  white-space: nowrap;
  overflow: hidden;
}
```

Not as big as a performance hit.

Comment (acacheung) [inline uxd/class-naming-conventions.md]: It doesn't actually work in all cases. We used that previously on mhe and voiceover did not read it.

**Pull request merged**